src/test/prettyPrint/expressions.spec.ts

```diff
@@ -0,0 +1,119 @@
+import fc from "fast-check";
+import {
+    AstConditional,
+    AstExpression,
+    AstNumber,
+    AstOpBinary,
+    AstOpUnary,
+    eqExpressions,
+    getAstFactory,
+} from "../../grammar/ast";
+import { prettyPrint } from "../../prettyPrinter";
+import { dummySrcInfo, getParser } from "../../grammar";
+
+describe("Pretty Print Expressions", () => {
+    // Max depth of the expression tree
+    const maxShrinks = 15;
+
+    const generateAstNumber = () =>
+        fc.record({
+            kind: fc.constant("number"),
+            base: fc.constantFrom(2, 8, 10, 16),
+            value: fc.bigInt().filter((n) => n > 0),
+            id: fc.constant(0),
+            loc: fc.constant(dummySrcInfo),
```

Comment:
I'd add a wrapper to automatically add these `id` and `loc` to the generated AST nodes

Reply:
All generators wrapped by `dummyAstNode` now